hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieMetadataWriteUtils.java

```diff
@@ -107,6 +107,7 @@ public static HoodieWriteConfig createMetadataWriteConfig(
 
     // Create the write config for the metadata table by borrowing options from the main write config.
     HoodieWriteConfig.Builder builder = HoodieWriteConfig.newBuilder()
+        .withIncrementalTableServiceEnable(false)
```

Comment:
Hi @danny0405 Maybe we need to disable incremental compaction for MDT. Since MDT compaction instant is not based on processing time but based on `https://github.com/apache/hudi/blob/dee6274e56d8b7d3852104d3e45377f4909466cd/hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java#L1576`

Which could cause selecting instants based on window is inaccurate, thus missing compaction

what do u think?

This is a screenshot of the debug process during the specific test.

<img width="1339" alt="截屏2025-01-17 15 46 49" src="https://github.com/user-attachments/assets/e9ae1edd-5d26-4df8-b6a0-7fc8c62e6d28" />

Reply:
Or we could deprecated right boundary which means [left-boundary, +∞)